tasks/seal/task_sdr.go

```diff
@@ -84,73 +80,11 @@ func (s *SDRTask) Do(taskID harmonytask.TaskID, stillOwned func() bool) (done bo
 	}
 	sectorParams := sectorParamsArr[0]
 
-	var pieces []struct {
-		PieceIndex  int64  `db:"piece_index"`
-		PieceCID    string `db:"piece_cid"`
-		PieceSize   int64  `db:"piece_size"`
-		DataRawSize *int64 `db:"data_raw_size"`
-	}
-
-	err = s.db.Select(ctx, &pieces, `
-		SELECT piece_index, piece_cid, piece_size, data_raw_size
-		FROM sectors_sdr_initial_pieces
-		WHERE sp_id = $1 AND sector_number = $2 ORDER BY piece_index ASC`, sectorParams.SpID, sectorParams.SectorNumber)
+	dealData, err := dealdata.DealDataSDRPoRep(ctx, s.db, s.sc, sectorParams.SpID, sectorParams.SectorNumber, sectorParams.RegSealProof)
```

Comment:
DealData will create readers for the pieces, we don't need that, just want the CommD.

Probably easiest addressed by adding a `onlyCommD` bool param to the `dealdata.DealDataSDRPoRep` func

Reply:
Adding this. We it would also reduce the footPrint if the function for sync task. It doesn't require data readers either.